code/modules/library/lib_machines.dm

```diff
@@ -432,17 +436,17 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
 				var/author = query.item[2]
 				var/title = query.item[3]
 				var/content = query.item[4]
-				var/obj/item/weapon/book/B = new(src.loc)
+				var/obj/item/weapon/book/B = new(get_turf(src))
 				B.name = "Book: [title]"
 				B.title = title
 				B.author = author
 				B.dat = content
 				B.icon_state = "book[rand(1,8)]"
-				src.visible_message("[src]'s printer hums as it produces a completely bound book. How did it do that?")
+				visible_message("[src]'s printer hums as it produces a completely bound book. How did it do that?")
 				break
-	src.add_fingerprint(usr)
-	src.updateUsrDialog()
-	return
+
+	add_fingerprint(usr)
+	//updateUsrDialog()

```

Comment:
It was already like that, I commented on accident(testing something) and forgot to uncomment.